includes/templates/class-amp-post-template.php

```diff
@@ -305,7 +305,7 @@ private function build_post_publish_timestamp() {
 		$format = 'U';
 
 		if ( empty( $this->post->post_date_gmt ) || '0000-00-00 00:00:00' === $this->post->post_date_gmt ) {
-			$timestamp = time();
+			$timestamp =  (int) get_the_time('U');
```

Comment:
The use of `time()` came from #5450. Why use `get_the_time()`?

For one thing, it's not passing the `$this->post` as context.